src/engine/Graphics/NineSlice.ts

```diff
@@ -0,0 +1,559 @@
+    this.canvasB = document.createElement('canvas');
+    this.canvasC = document.createElement('canvas');
+    this.canvasD = document.createElement('canvas');
+    this.canvasE = document.createElement('canvas');
+    this.canvasF = document.createElement('canvas');
+    this.canvasG = document.createElement('canvas');
+    this.canvasH = document.createElement('canvas');
+    this.canvasI = document.createElement('canvas');
+    this.initialize();
+
+    if (!this.imgSource.isLoaded()) {
+      this._logger.warnOnce(
+        `ImageSource ${this.imgSource.path}` +
+          ` is not yet loaded and won't be drawn. Please call .load() or include in a Loader.\n\n` +
+          `Read https://excaliburjs.com/docs/imagesource for more information.`
+      );
+    }
+  }
+
+  /**
+   *  Sets the target width of the 9 slice (pixels), and recalculates the 9 slice if desired (auto)
+   */
+  setTargetWidth(newWidth: number, auto: boolean = false) {
+    this.config.width = newWidth;
+    if (auto) {
+      this.initialize();
+    }
+  }
+
+  /**
+   *  Sets the target height of the 9 slice (pixels), and recalculates the 9 slice if desired (auto)
+   */
+  setTargetHeight(newHeight: number, auto: boolean = false) {
+    this.config.height = newHeight;
+    if (auto) {
+      this.initialize();
+    }
+  }
+
+  /**
+   *  Sets the 9 slice margins (pixels), and recalculates the 9 slice if desired (auto)
+   */
+  setMargins(left: number, top: number, right: number, bottom: number, auto: boolean = false) {
+    this.config.sourceConfig.leftMargin = left;
+    this.config.sourceConfig.topMargin = top;
+    this.config.sourceConfig.rightMargin = right;
+    this.config.sourceConfig.bottomMargin = bottom;
+    if (auto) {
+      this.initialize();
+    }
+  }
+
+  /**
+   *  Sets the stretching strategy for the 9 slice, and recalculates the 9 slice if desired (auto)
+   *
+   */
+  setStretch(type: 'Horizontal' | 'Vertical' | 'Both', strategy: NineSliceStretch, auto: boolean = false) {
+    if (type === 'Horizontal') {
+      this.config.destinationConfig.stretchH = strategy;
+    } else if (type === 'Vertical') {
+      this.config.destinationConfig.stretchV = strategy;
+    } else {
+      this.config.destinationConfig.stretchH = strategy;
+      this.config.destinationConfig.stretchV = strategy;
+    }
+    if (auto) {
+      this.initialize();
+    }
+  }
+
+  /**
+   *  Returns the config of the 9 slice
+   */
+  getConfig(): NineSliceConfig {
+    return this.config;
+  }
+
+  /**
+   *  Draws 1 of the 9 tiles based on parameters passed in
+   *  context is the ExcaliburGraphicsContext from the _drawImage function
+   *  destinationSize is the size of the destination image as a vector (width,height)
+   *  targetCanvas is the canvas to draw to
+   *  hstrategy and vstrategy are the horizontal and vertical stretching strategies
+   *  marginW and marginH are optional margins for the 9 slice for positioning
+   */
+  _drawTile(
```

Comment:
Should this be part of the public API?

Reply:
nope, both _drawTile and _calculateParams are internal utility functions

Reply:
see other comment https://github.com/excaliburjs/Excalibur/pull/3241#discussion_r1797640168

Reply:
fixed